### PR TITLE
Feature/update to polkadot v0 8 30

### PR DESCRIPTION
**Pull request by MRamanenkau**

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -1253,10 +1283,10 @@ impl_runtime_apis! {
 			sp_consensus_babe::BabeGenesisConfiguration {
 				slot_duration: Babe::slot_duration(),
 				epoch_length: EpochDuration::get(),
-				c: PRIMARY_PROBABILITY,
+				c: BABE_GENESIS_EPOCH_CONFIG.c,
 				genesis_authorities: Babe::authorities(),
 				randomness: Babe::randomness(),
-				allowed_slots: sp_consensus_babe::AllowedSlots::PrimaryAndSecondaryPlainSlots,
+				allowed_slots: BABE_GENESIS_EPOCH_CONFIG.allowed_slots,
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:1274]: All constants values stated the same.
[Substrate PR](https://github.com/paritytech/substrate/pull/8072/files#diff-67684005af418e25ff88c2ae5b520f0c040371f1d817e03a3652e76b9485224aR122-R128)

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -1192,7 +1222,7 @@ impl_runtime_apis! {
 		}
 
 		fn random_seed() -> <Block as BlockT>::Hash {
-			RandomnessCollectiveFlip::random_seed()
+			pallet_babe::RandomnessFromOneEpochAgo::<Runtime>::random_seed().0
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:1210]: [Refactoring](https://github.com/paritytech/substrate/pull/8180)

File: bin/node-template/pallets/template/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+//! Benchmarking setup for pallet-template
```
Comment (AndreiNavoichyk) [inline bin/node-template/pallets/template/src/benchmarking.rs]: We don't need node template

File: bin/node/runtime/Cargo.toml [hunk 4]
```diff
@@ -36,10 +37,11 @@ sp-keyring = { version = "3.0.0", optional = true, path = "../../../primitives/k
 sp-session = { version = "3.0.0", default-features = false, path = "../../../primitives/session" }
 sp-transaction-pool = { version = "3.0.0", default-features = false, path = "../../../primitives/transaction-pool" }
 sp-version = { version = "3.0.0", default-features = false, path = "../../../primitives/version" }
+sp-npos-elections = { version = "3.0.0", default-features = false, path = "../../../primitives/npos-elections" }
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/Cargo.toml:40]: What does this pallet responsible for?

File: bin/node/runtime/src/lib.rs [hunk 5]
```diff
@@ -120,11 +120,18 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	// implementation changes and behavior does not, then leave spec_version as
 	// is and increment impl_version.
 	spec_version: 284,
-	impl_version: 0,
+	impl_version: 1,
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs]: I see the pallets major versions increase. It should be spec_version increase in the runtime

File: bin/node/runtime/src/lib.rs [hunk 6]
```diff
@@ -467,17 +474,11 @@ parameter_types! {
...
-	// 0.05%. The higher the value, the more strict solution acceptance becomes.
-	pub MinSolutionScoreBump: Perbill = Perbill::from_rational_approximation(5u32, 10_000);
-	pub OffchainSolutionWeightLimit: Weight = RuntimeBlockWeights::get()
-		.get(DispatchClass::Normal)
-		.max_extrinsic.expect("Normal extrinsics have a weight limit configured; qed")
-		.saturating_sub(BlockExecutionWeight::get());
 }
 
 impl pallet_staking::Config for Runtime {
+	const MAX_NOMINATIONS: u32 =
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:480]: How will it affect the nominators amount?

File: bin/node/runtime/src/lib.rs [hunk 7]
```diff
@@ -495,17 +496,9 @@ impl pallet_staking::Config for Runtime {
 		pallet_collective::EnsureProportionAtLeast<_3, _4, AccountId, CouncilCollective>
 	>;
 	type SessionInterface = Self;
-	type RewardCurve = RewardCurve;
+	type EraPayout = pallet_staking::ConvertCurve<RewardCurve>;
```
Comment (AndreiNavoichyk) [inline bin/node/runtime/src/lib.rs:499]: We need to check payouts

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:1274]: Could you add summary here?

Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:1274]: The Babe runtime module has [the current_epoch() and next_epoch() functions](https://github.com/paritytech/substrate/blob/93b231e79f5b4e551c34234e89fa4a2e5e9c1510/primitives/consensus/babe/src/lib.rs#L381-L386) that return the current and next epoch. 
Before the changes, they hadn't given any information about the configuration, which can potentially change between epochs. This changes fix this.

File: bin/node/runtime/Cargo.toml (hunk 4 above)
Comment (MRamanenkau) [reply bin/node/runtime/Cargo.toml:40]: This pallet contains a set of election algorithms and is used for making the number of nominations configurable

File: bin/node/runtime/src/lib.rs (hunk 6 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:480]: It doesn't affect nominators amount. This variable set how many nominations of validators can select one nominator.
So now, one nominator can vote for 16 different validators. And far as I understand, this number stayed the same. Can you confirm it?

File: bin/node/runtime/src/lib.rs (hunk 7 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:499]: Let's do it together

File: bin/node/runtime/src/lib.rs (hunk 5 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs]: spec_version incremented. impl_version reset

File: bin/node-template/pallets/template/src/benchmarking.rs (hunk 3 above)
Comment (MRamanenkau) [reply bin/node-template/pallets/template/src/benchmarking.rs]: Removed

File: bin/node/runtime/src/lib.rs (hunk 6 above)
Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:480]: We should test it.

File: bin/node/runtime/Cargo.toml (hunk 4 above)
Comment (AndreiNavoichyk) [reply bin/node/runtime/Cargo.toml:40]: In what version Polkadot did add it?

Comment (MRamanenkau) [reply bin/node/runtime/Cargo.toml:40]: 0.8.30

**Review by AndreiNavoichyk — DISMISSED** [review]

File: bin/node/runtime/src/lib.rs (hunk 6 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:480]: Now the amount of nominations is set [here](https://github.com/Cerebellum-Network/pos-network-node/pull/141/files#diff-67684005af418e25ff88c2ae5b520f0c040371f1d817e03a3652e76b9485224aR528)
Previosly it was set [here](https://github.com/Cerebellum-Network/pos-network-node/blob/feature/update-to-polkadot-v0-8-29/frame/staking/src/lib.rs#L380)
The constant value stayed the same(16)

Comment (AndreiNavoichyk) [reply bin/node/runtime/src/lib.rs:480]: Great! Thank you @MRamanenkau

File: bin/node/runtime/src/lib.rs (hunk 7 above)
Comment (MRamanenkau) [reply bin/node/runtime/src/lib.rs:499]: Tested. Validators are able to get their payouts.

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**